### PR TITLE
Allow backuplocation to be passed in for migration 

### PR DESCRIPTION
**Pull request by ram-infrac**
**What type of PR is this?**
>feature

**What this PR does / why we need it**:
Allow using backuplocation for migration operation
  - accept backuplocation as storage option for clusterpair
  - create backuplocation cr on destination cluster
  - pass backuplocation to portworx driver clusterpair api
 -  update openstorage vendor
**Does this PR change a user-facing CRD or CLI?**:
no
**Is a release note needed?**:
no

**Does this change need to be cherry-picked to a release branch?**:
2.5

### CODE REVIEW

File: drivers/volume/portworx/portworx.go [hunk 1]
```diff
@@ -2125,11 +2125,16 @@ func (p *portworx) CreatePair(pair *storkapi.ClusterPair) (string, error) {
 		return "", fmt.Errorf("cannot get cluster manager, err: %s", err.Error())
 	}
 
+	var credID string
+	if bkpl, ok := pair.Spec.Options["backuplocation"]; ok {
```
[the remark this answers is not in the document]
Comment (ram-infrac) [reply drivers/volume/portworx/portworx.go]: yes we can use stork_api backuplocationname used for cr registration

File: pkg/migration/controllers/clusterpair.go [hunk 2]
```diff
@@ -147,18 +150,72 @@ func (c *ClusterPairController) handle(ctx context.Context, clusterPair *stork_a
...
+					err.Error())
+				return c.client.Update(context.TODO(), clusterPair)
+			}
+			resp.ResourceVersion = ""
+			if _, err := remoteClient.CreateBackupLocation(resp); err != nil && !errors.IsAlreadyExists(err) {
+				c.recorder.Event(clusterPair,
+					v1.EventTypeWarning,
+					string(clusterPair.Status.SchedulerStatus),
+					err.Error())
+				return c.client.Update(context.TODO(), clusterPair)
```
[the remark this answers is not in the document]
Comment (ram-infrac) [reply pkg/migration/controllers/clusterpair.go]: yes, make sense . move to func. thanks

File: pkg/migration/controllers/clusterpair.go [hunk 3]
```diff
@@ -220,3 +230,45 @@ func (c *ClusterPairController) createCRD() error {
...
+			if err != nil && !errors.IsAlreadyExists(err) {
+				return err
+			}
+		}
+		// create backuplocation on destination cluster
+		resp, err := storkops.Instance().GetBackupLocation(bkpl, clusterPair.GetNamespace())
+		if err != nil {
+			return err
+		}
+		resp.ResourceVersion = ""
```
Comment (disrani-px) [inline pkg/migration/controllers/clusterpair.go:268]: Don't you also have to remove the UUID, self link and other members from the metadata?

File: pkg/migration/controllers/clusterpair.go [hunk 4]
```diff
@@ -220,3 +230,45 @@ func (c *ClusterPairController) createCRD() error {
 
 	return apiextensions.Instance().ValidateCRD(resource, validateCRDTimeout, validateCRDInterval)
 }
+
+func createBackupLocationOnRemote(remoteConfig *restclient.Config, clusterPair *stork_api.ClusterPair) error {
```
Comment (disrani-px) [inline pkg/migration/controllers/clusterpair.go:234]:
```suggestion
func (c *ClusterPairController) createBackupLocationOnRemote(remoteConfig *restclient.Config, clusterPair *stork_api.ClusterPair) error {
```

File: drivers/volume/portworx/portworx.go [hunk 5]
```diff
@@ -2125,11 +2125,16 @@ func (p *portworx) CreatePair(pair *storkapi.ClusterPair) (string, error) {
 		return "", fmt.Errorf("cannot get cluster manager, err: %s", err.Error())
 	}
 
+	var credID string
+	if bkpl, ok := pair.Spec.Options[storkapi.BackupLocationResourceName]; ok {
```
Comment (disrani-px) [inline drivers/volume/portworx/portworx.go:2129]: Add a version check for 2.6.0 here and return an error if it is an older version

File: pkg/migration/controllers/clusterpair.go (hunk 3 above)
Comment (ram-infrac) [reply pkg/migration/controllers/clusterpair.go:268]: only  setting `resourceVersion` restrict resource creation on remote cluster

**Pull request merged**